src/Faulkner/avoidance.py

```diff
@@ -5,65 +5,81 @@
 import loghelper
 
 class avoidance:
-    def __init__(self, state_vars, params):
+    def __init__(self, state_vars, params, history):
         self.state_vars = state_vars
         self.params = params
+        self.history = history
     
     #def is_obstructed(self):
     #    return self.front() or self.fleft() or self.frontAndFright() or self.frontAndFleft() or self.frontAndFleftAndFright()
         #return self.front() or self.frontAndFright() or self.frontAndFleft() or self.frontAndFleftAndFright()
 
-    def is_obstructed(self):
-        return self.state_vars.regions['front'] <= self.params.avoid_distance or self.state_vars.regions['fleft'] <= self.params.avoid_distance or self.state_vars.regions['fright'] <= self.params.avoid_distance
+    #def is_obstructed(self):
+    #    return self.state_vars.regions['front'] <= self.params.avoid_distance or self.state_vars.regions['fleft'] <= self.params.avoid_distance or self.state_vars.regions['fright'] <= self.params.avoid_distance
+    def obstacle_near_centre(self, regions):
+        return regions['front'] <= self.params.avoid_distance or regions['fleft'] <= self.params.avoid_distance or regions['fright'] <= self.params.avoid_distance
     
+    def is_obstructed(self, regions):
+        self.regions = regions
+        history_window = self.history.getLastN(self.params.is_obstructed_window)
+        obs = False
+        for it in history_window:
+            if self.obstacle_near_centre(it.regions):
+                obs = True
+                break
+        return obs or self.obstacle_near_centre(self.regions)
```

Comment:
At what frequency does this function get called? I want to say the history_window code is not needed. Justification: this function gets called fast enough such that self.obstacle_near_centre gets called in time, so the history window doesn't hold any unique/relevant info.